src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs

```diff
@@ -334,26 +334,45 @@ public override void Write(char[] buffer)
 
             CheckAsyncTaskInProgress();
 
-            int index = 0;
             int count = buffer.Length;
-            while (count > 0)
+			
+            // Threshold of 4 was chosen after running perf tests
+            if (count <= 4)
             {
-                if (_charPos == _charLen)
+                for (int i = 0; i < count; i++)
```

Comment:
Minor, but I wonder if the JIT will eliminate the bounds check on buffer[i] with this, and if not, whether it would if this were changed to `i < buffer.Length` instead of `i < count`.